test/routes/src/server/routes/t_launchpad.js

```diff
@@ -569,6 +570,214 @@ describe('The Launchpad API endpoint', () => {
       });
     });
   });
+
+
+  describe('get snap builds route', () => {
+    const lp_snap_user = 'test-snap-user';
+    const lp_snap_name = 'test-snap-name';
+
+    let lp_api_url;
+    let lp_snap_path;
+    let lp_builds_path;
+    let lp_snap_link;
+
+    before(() => {
+      lp_api_url = conf.get('LP_API_URL');
+      lp_snap_path = `/devel/~${lp_snap_user}/+snap/${lp_snap_name}`;
+      lp_builds_path = `${lp_snap_path}/builds`;
+
+      lp_snap_link = `${lp_api_url}${lp_snap_path}`;
+    });
+
+    context('when snap and builds are successfully fetched', () => {
+
+      beforeEach(() => {
+        // when getting snap data from API (via self_link)
+        nock(lp_api_url)
+          .get(lp_snap_path)
+          .reply(200, {
+            name: lp_snap_name,
+            builds_collection_link: `${lp_api_url}${lp_builds_path}`
+          });
+
+        // when getting builds list (via builds_collection_link)
+        nock(lp_api_url)
+          .get(lp_builds_path)
+          .query({
+            'ws.start': 0,
+            'ws.size': 10
+          })
+          .reply(200, {
+            total_size: 1,
+            start: 0,
+            entries: [
+              {
+                'self_link': `${lp_api_url}${lp_snap_path}/+build/12345`,
+                'date_started': '2016-12-06T10:40:17.751937+00:00',
+                'datebuilt': '2016-12-06T10:47:59.384231+00:00',
+                'duration': '0:07:41.632294',
+                'datecreated': '2016-12-06T10:30:29.285058+00:00',
+                'buildstate': 'Successfully built',
+                'resource_type_link': `${lp_api_url}/devel/#snap_build`,
+                'build_log_url': `${lp_api_url}${lp_snap_path}/+build/12345/+files/buildlog_snap_ubuntu_xenial_armhf_${lp_snap_name}_BUILDING.txt.gz`,
+                'arch_tag': 'armhf',
+              }
+            ]
+          });
+      });
+
+      afterEach(() => {
+        nock.cleanAll();
+      });
+
+      it('should return a 200 OK response', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(200, done);
+      });
+
+      it('should return a "success" status', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(hasStatus('success'))
+          .end(done);
+      });
+
+      it('should return body with "snap-builds-found" message', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .expect(hasMessage('snap-builds-found'))
+          .end(done);
+      });
+
+      it('should return builds list as payload message', (done) => {
+        supertest(app)
+          .get('/launchpad/builds')
+          .query({ snap_link: lp_snap_link })
+          .end((err, res) => {
+            expect(res.body.payload.message.length).toEqual(1);
```

Comment:
Can the content of the builds list and the statusCode be tested as well? The endpoint could return a 404 and an error payload and I _think_ this may report a false positive.